src/three-components/SmoothControls.ts

```diff
@@ -0,0 +1,533 @@
+/*
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the 'License');
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Camera, EventDispatcher, Quaternion, Spherical, Vector2, Vector3} from 'three';
+
+import {clamp, step} from '../utils.js';
+
+export type EventHandlingBehavior = 'prevent-all'|'prevent-handled';
+export type ZoomPolicy = 'always-allow'|'allow-when-focused';
+export type TouchMode = 'rotate'|'zoom';
+
+export interface SmoothControlsConstraints {
+  nearOrbitRadius?: number;
+  farOrbitRadius?: number;
```

Comment:
`near` and `far` imply (IMO) that it has something to do with the camera's near and far planes and was unclear what these values did until digging more -- maybe, similar to the polar/azimuthal config, and using terminology used elsewhere in the controls (`zoom` and `radius`), maybe something like `minimumZoom`/`maximumZoom`, `minimumRadius`, `maxiumRadius`

Reply:
I haven't fully thought this out, but it seems like `zoom` and `radius` are distinct things and that it is better to be explicit about what is being adjusted.

I agree that `near` and `far` are weird words to use, can change that to `minimum` and `maximum`.

Reply:
Looking again, agreed that `zoom` doesn't match -- `minimumRadius` and `maximumRadius` sound great to me and inline with the other options

Reply:
Done